.github/workflows/main.yaml

```diff
@@ -21,6 +21,7 @@ jobs:
     # variables in context environments are setup in github to match branch names
     environment:
       name: ${{ github.ref_name }}
+      ACTIONS_ALLOW_USE_UNSECURE_NODE_VERSION: true
```

Comment:
sorry, i mislead you in my previous suggestion.  I thought `env` and `environment` are the same thing, but in actually they are completely different.  please take a look at the [GH action docs](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#defining-environment-variables-for-a-single-workflow) and notice that it's `env`.  You'll need to define an `env:` at the job level and add `ACTIONS_ALLOW_USE_UNSECURE_NODE_VERSION: true` to it.

Reply:
nope, this won't work.  The step that's failing is `actions/checkout@v4` which is in the `test` job, this however is scoped to the `deploy` job.  I would move `env:` to the same level as `on:` and `jobs:` making scoped globally in this workflow.